app/scripts/controllers/centers/ManageGroupMembersController.js

```diff
@@ -1,20 +1,41 @@
 (function (module) {
     mifosX.controllers = _.extend(module, {
-        ManageGroupMembersController: function (scope, resourceFactory, location, routeParams, $modal) {
+        ManageGroupMembersController: function (scope, resourceFactory, $http, $rootScope, API_VERSION, location, routeParams, $modal) {
         	
         	scope.centerId = routeParams.id;
             scope.indexOfClientToBeDeleted = "";
+            scope.resultList = [];
 
             scope.viewGroup = function (item) {
                 scope.group = item;
             };
 
-            resourceFactory.centerResource.get({centerId: routeParams.id, template: 'true', associations: 'groupMembers'}, function (data) {
+            resourceFactory.centerResource.get({centerId: routeParams.id, template: 'true',associations: 'groupMembers'}, function (data) {
                 scope.data = data;
-                scope.groupsOptions = data.groupMembersOptions;
                 scope.groups = data.groupMembers;
             });
-            
+
+            scope.groupsOptions = function(value){
+                return $http({
+                    method: 'GET',
+                    url: $rootScope.hostUrl + API_VERSION + '/groups' + '?name=' + value + '&orderBy=' + 'name' + '&orphansOnly=' + 'true'
+                    + '&officeId=' + scope.data.officeId + '&sortOrder=' + 'ASC',
+                    data: {}
+                }).then(function (response) {
+                    if(response != ""){
+                        scope.resultList = response.data;
+                        for(var i in scope.data.groupMembers){
```

Comment:
why do you need these two loop ? The API returns only those groups which are not associated with any center